src/core/agent/services/connectionService.ts

```diff
@@ -252,10 +270,81 @@ class ConnectionService extends AgentService {
 
   @OnlineOnly
   async deleteConnectionById(id: string): Promise<void> {
-    await this.props.signifyClient.contacts().delete(id);
+    await this.props.signifyClient
+      .contacts()
+      .delete(id)
+      .catch((error) => {
+        const status = error.message.split(" - ")[1];
+        if (!/404/gi.test(status)) {
+          throw error;
+        }
+      });
     await this.connectionStorage.deleteById(id);
   }
 
+  async markConnectionPendingDelete(id: string) {
+    const connectionProps = await this.connectionStorage.findById(id);
+    if (!connectionProps) return;
+    connectionProps.pendingDeletion = true;
+    await this.connectionStorage.update(connectionProps);
+
+    this.props.eventEmitter.emit<ConnectionRemovedEvent>({
+      type: EventTypes.ConnectionRemoved,
+      payload: {
+        connectionId: id,
+      },
+    });
+  }
+
+  async getConnectionsPendingDeletion() {
+    const connections = await this.connectionStorage.findAllByQuery({
+      pendingDeletion: true,
+    });
+
+    return connections.map((connection) => connection.id);
+  }
+
+  async getConnectionsPending() {
+    const connections = await this.connectionStorage.findAllByQuery({
+      pending: true,
+      groupId: undefined,
+    });
+
+    return connections.map((connection) => connection.id);
+  }
+
+  @OnlineOnly
+  async reconnectPendingConnectionById(id: string) {
+    const connection = await this.props.signifyClient
+      .contacts()
+      .get(id)
```

Comment:
Why do we still have this call? As I said, the contact won't exist before we've finished resolving the OOBI. You need to embed the OOBI URL in the pending connection record.

To me this seems like you haven't tested this flow properly locally. You need to setup your environment in such a way that you're testing is when developing it rather than guessing.